semeio/jobs/correlated_observations_scaling/scaled_matrix.py

```diff
@@ -3,15 +3,17 @@
 import numpy as np
 
 from semeio.jobs.correlated_observations_scaling.exceptions import EmptyDatasetException
+from semeio.reporter import Reporter
 
 
 class DataMatrix(object):
-    def __init__(self, input_data):
+    def __init__(self, input_data, reporter):
```

Comment:
I think it is silly that a `DataMatrix` should be initialized with a `reporter`. I wonder if `Reporter` should be implemented like `logging`?

Reply:
I think that is a good observation ;)

Reply:
I would say that if `DataMatrix` wants to post structured data, then it should not be allowed to (and we would have to pass the `reporter` as you've done). Then I think we should redesign the code.

However, for general logging statements it could be an idea to register a listener to the Python logger and hence fetch all logging statements and pass them on via the reporter 🤷